File: report-viewer/src/model/Match.ts
/**
 * Match between two files of two submissions.
 * @property firstFile - Path to the file of the first submission.
 * @property secondFile - Path to the file of the second submission.
 * @property startInFirst - Starting line of the match in the first file.
 * @property endInFirst - Ending line of the match in the first file.
 * @property startInSecond - Starting line of the match in the second file.
 * @property endInSecond - Ending line of the match in the second file.
 * @property tokens - Number of tokens in the match.
 * @property colorIndex - Index of the color to use for the match.
 */
export interface Match {
  firstFile: string
  secondFile: string
  startInFirst: CodePosition
  endInFirst: CodePosition
  startInSecond: CodePosition
  endInSecond: CodePosition
  tokens: number
  colorIndex?: number
}

export interface CodePosition {
  line: number
  column: number
  tokenListIndex: number
}

File: reporting/src/main/java/de/jplag/reporting/reportobject/model/Match.java
package de.jplag.reporting.reportobject.model;

/**
 * Represents a matched code segment between two files. Stores metadata about the matching code positions in both files.
 * @param firstFileName Name of the first file involved in the match
 * @param secondFileName Name of the second file involved in the match
 * @param startInFirst Start position of the match in the first file
 * @param endInFirst End position of the match in the first file
 * @param startInSecond Start position of the match in the second file
 * @param endInSecond End position of the match in the second file
 * @param lengthOfFirst Length of the matched segment in the first file
 * @param lengthOfSecond Length of the matched segment in the second file
 */
public record Match(String firstFileName, String secondFileName, CodePosition startInFirst, CodePosition endInFirst, CodePosition startInSecond,
        CodePosition endInSecond, int lengthOfFirst, int lengthOfSecond) {
}
